assessment_module_manager/assessment_module_manager/module/request_to_module.py

```diff
@@ -32,23 +34,37 @@ async def find_module_by_name(
     return None
 
 
-async def request_to_module(module: Module, path: str, data: dict) -> ModuleResponse:
+async def request_to_module(module: Module, module_config: Optional[str], path: str, data: Optional[dict], method: str) -> ModuleResponse:
     """
     Helper function to send a request to a module.
     It raises appropriate FastAPI HTTPException if the request fails.
     """
     module_secret = env.MODULE_SECRETS[module.name]
+    headers = {'X-API-Secret': module_secret} if module_secret else None
+    if module_config:
+        if headers is None:
+            headers = {}
+        headers['X-Module-Config'] = module_config
+
     try:
         async with httpx.AsyncClient(base_url=module.url, timeout=600) as client:
-            response = await client.post(path, json=data, headers={'X-API-Secret': module_secret} if module_secret else None)
+            if method == "POST":
+                response = await client.post(path, json=data, headers=headers)
+            else:
+                response = await client.get(path, headers=headers)
```

Comment:
```suggestion
            if method == "POST":
                response = await client.post(path, json=data, headers=headers)
            elif method == "GET":
                response = await client.get(path, headers=headers)
            else:
                raise ...
```
`PUT` and `DELETE` requests are also possible. These would currently just get converted into `GET` requests.